apps/comps/rpc/middleware/cache.ts

```diff
@@ -0,0 +1,70 @@
+import { StandardRPCJsonSerializer } from "@orpc/client/standard";
+import {
+  Context,
+  ErrorMap,
+  Meta,
+  Middleware,
+  ORPCErrorConstructorMap,
+  os,
+} from "@orpc/server";
+import { unstable_cache } from "next/cache";
+import type { Logger } from "pino";
+
+export function cacheMiddleware<
+  TInContext extends Context & { logger: Logger },
+  TInput,
+  TOutput,
+  TErrorMap extends ErrorMap,
+  TMeta extends Meta,
+>(options?: {
+  key?: string[];
+  revalidate?: number;
+  tags?: string[];
+  includeContext?: (context: TInContext) => Record<string, any>;
```

Comment:
This extra arg is a callback that allows the caller to select data from the context that should be included in the cache key. This is useful when the procedure input doesn't fully encompass what the cache key should be. For example, if a procedure uses auth middleware and the user id from the context, and passes the user id along with procedure input into a service function, you'd want to select the user id from the context so it is included in the cache key.

Reply:
lets document this behavior please!